packages/neutron-sdk/src/interchain_queries/register_queries.rs

```diff
@@ -68,6 +70,120 @@ pub fn register_balance_query_msg(
+/// * **denom** denomination of the coin for which you want to get balance;
+/// * **update_period** is used to say how often the query must be updated.
+pub fn register_bank_total_query_msg(
+    deps: DepsMut<InterchainQueries>,
+    env: Env,
+    connection_id: String,
+    denom: String,
+    update_period: u64,
+) -> NeutronResult<NeutronMsg> {
+    let supply_key = create_total_denom_key(denom)?;
+
+    let kv_key = KVKey {
+        path: BANK_STORE_KEY.to_string(),
+        key: Binary(supply_key),
+    };
+
+    register_interchain_query_msg(
+        deps,
+        env,
+        connection_id,
+        QueryType::KV,
+        vec![kv_key],
+        String::new(),
+        update_period,
+    )
+}
+
+/// Creates a message to register an Interchain Query to get fee pool on remote chain from distribution module
+///
+/// * **connection_id** is an IBC connection identifier between Neutron and remote chain;
+/// * **update_period** is used to say how often the query must be updated.
+pub fn register_distribution_fee_pool_query_msg(
+    deps: DepsMut<InterchainQueries>,
+    env: Env,
+    connection_id: String,
+    update_period: u64,
+) -> NeutronResult<NeutronMsg> {
+    let kv_key = KVKey {
+        path: DISTRIBUTION_STORE_KEY.to_string(),
+        key: Binary(create_fee_pool_key()?),
+    };
+
+    register_interchain_query_msg(
+        deps,
+        env,
+        connection_id,
+        QueryType::KV,
+        vec![kv_key],
+        String::new(),
+        update_period,
+    )
+}
+
+/// Creates a message to register an Interchain Query to get governance proposal on remote chain
+///
+/// * **connection_id** is an IBC connection identifier between Neutron and remote chain;
+/// * **proposal_id** is a proposal id from remote chain.
+/// * **update_period** is used to say how often the query must be updated.
+pub fn register_gov_proposal_query_msg(
+    deps: DepsMut<InterchainQueries>,
+    env: Env,
+    connection_id: String,
+    proposal_id: u64,
+    update_period: u64,
+) -> NeutronResult<NeutronMsg> {
+    let kv_key = KVKey {
+        path: GOV_STORE_KEY.to_string(),
+        key: Binary(create_gov_proposal_key(proposal_id)?),
+    };
+
+    register_interchain_query_msg(
+        deps,
+        env,
+        connection_id,
+        QueryType::KV,
+        vec![kv_key],
+        String::new(),
+        update_period,
+    )
+}
+
+/// Creates a message to register an Interchain Query to get validator info on remote chain
+///
+/// * **connection_id** is an IBC connection identifier between Neutron and remote chain;
+/// * **validator** is an validator operator address of an account on remote chain for which you want to get rewards ;
+/// * **update_period** is used to say how often the query must be updated.
+pub fn register_staking_validator_query_msg(
+    deps: DepsMut<InterchainQueries>,
+    env: Env,
+    connection_id: String,
+    validator: String,
```

Comment:
And the same about validators. It's more convenient to query multiple validators in one query.